Program.cs

```diff
@@ -981,22 +1048,30 @@ internal bool End()
             TypeWriter.WriteLine();
             TypeWriter.WriteLine($"Final score: {totalXP}");
             TypeWriter.WriteLine();
-            TypeWriter.WriteLine("Game Over",TypeWriter.Speed.Talk);    
-            TypeWriter.WriteLine($"psst {cottonUserName} do you want to try again: yes / no",TypeWriter.Speed.Talk);
-            
-            string tryAgain = Console.ReadLine();
-            switch(tryAgain.ToLower())
-            {
-                case "yes":
-                    playerReset();
-                    return true;
+            TypeWriter.WriteLine("Game Over",TypeWriter.Speed.Talk);
 
-                default:
-                    TypeWriter.WriteLine("bye for now :)", TypeWriter.Speed.Talk);
-                    TypeWriter.WriteLine("");
-                    UpdateLeaderBoard(totalXP);
-                    return false;
+            if (isGameCorrupted == false)
+            {   
+                TypeWriter.WriteLine($"psst {cottonUserName} do you want to try again: yes / no",TypeWriter.Speed.Talk);
+                
+                string tryAgain = Console.ReadLine();
+                switch(tryAgain.ToLower())
+                {
+                    case "yes":
+                        playerReset();
+                        return true;
+
+                    default:
+                        TypeWriter.WriteLine("bye for now :)", TypeWriter.Speed.Talk);
+                        TypeWriter.WriteLine("");
+                        UpdateLeaderBoard(totalXP);
+                        return false;
+                }
             }
+            else
+            {
+                return false;
```

Comment:
Seems unfair that you don't get added to the leaderboard if the game gets corrupted ... but your call .. it will definitely make the player think.

Reply:
the player now still gets put on the leaderboard I think the game getting corrupted still some food for thought